signature/src/keypair.rs

```diff
@@ -1,17 +1,29 @@
 //! Signing keypairs.
 
-use crate::Signature;
-
 /// Signing keypair with an associated verifying key.
 ///
 /// This represents a type which holds both a signing key and a verifying key.
-pub trait Keypair<S: Signature>: AsRef<Self::VerifyingKey> {
+pub trait Keypair {
```

Comment:
Another maybe silly question: how is this intended to be used? It seems that the only thing `Keypari` is good for is getting a verification key. But you can't even do anything with that verification key, because it's not necessarily part of or an input to a `Verifier`. So when does this trait get used?

Reply:
`Keypair` originally started out with mandatory bounds for `Signer` and `Verifier`. We removed these because there are other possible traits `Keypair` could be used in conjunction with, like `DigestSigner`/`DigestVerifier`, `PrehashSigner`/`PrehashVerifier`, as well as `RandomizedSigner`/`RandomizedDigestSigner`/`RandomizedPrehashSigner`. See #1107.

So, like with `SignatureEncoding`, the idea is to provide more flexible traits that don't mandate bounds but are intended to be composed.

In the case of `Keypair`, the trait is intended only as a way to obtain a verifying key from a signing key.

See also #1124 /cc @lumag